modules/exploits/linux/http/atutor_filemanager_traversal.rb

```diff
@@ -0,0 +1,370 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'ATutor 2.2.1 Directory Traversal / Remote Code Execution',
+      'Description'    => %q{
+         This module exploits a directory traversal vulnerability in ATutor on an Apache/PHP
+         setup with display_errors set to On, which can be used to allow us to upload a malicious
+         ZIP file. On the web application, a blacklist verification is performed before extraction,
+         however it is not sufficient to prevent exploitation.
+
+         You are required to login to the target to reach the vulnerability, however this can be
+         done as a student account and remote registration is enabled by default.
+
+         Just incase remote registration isnt enabled, this module uses 2 vulnerabilities
+         in order to bypass the authenication:
+
+         1. confirm.php Authentication Bypass Type Juggling vulnerability
+         2. password_reminder.php Remote Password Reset TOCTOU vulnerability
+
+         ~ spirit of the hack
```

Comment:
Anyone using msf should have enough spirit of hack already that we can safely remove this line :)